api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -346,6 +350,77 @@ union OutboundType {
   loadBalancer: "loadBalancer",
 }
 
+/** The configuration that the operators of the cluster have to authenticate to Azure. */
+model OperatorsAuthenticationProfile {
+  /** Represents the information related to Azure User-Assigned managed identities needed
+   * to perform Operators authentication based on Azure User-Assigned Managed Identities */
+  managedIdentities: ManagedIdentitiesProfile;
+}
+
+/** Represents the information related to Azure User-Assigned managed identities needed
+ * to perform Operators authentication based on Azure User-Assigned Managed Identities */
+model ManagedIdentitiesProfile {
+  /** The set of Azure User-Assigned Managed Identities leveraged for the Control Plane
+   * operators of the cluster. The set of required managed identities is dependent on the
+   * Cluster's OpenShift version. */
+  @OpenAPI.extension("x-ms-identifiers", ["operatorName"])
+  controlPlaneOperators: ControlPlaneOperatorsElement[];
```

Comment:
We want to define these as a map type because otherwise replacement is difficult.  it'll go through additional scrutiny but we already have the use case built for classic.

Reply:
The way to do that in TypeSpec, as I understand, is with `Record<Type>` which defines a map with string keys and `Type` values.  But the TypeSpec compiler rejects it with
```
error @azure-tools/typespec-azure-resource-manager/arm-no-record: Model properties or
operation parameters should not be of type Record. ARM requires Resource provider teams
to define types explicitly.
```
so I'll have to look for a way to override it.

Reply:
Figured it out...
```
#suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "operator name to resource ID pairings"
controlPlaneOperators: Record<UserAssignedIdentityResourceId>;
```

Reply:
Fixed in a `fixup!` commit that I'll eventually squash.